### PR TITLE
fix: fixed upload of remote files to local storage

### PR DESCRIPTION
**Pull request by adubovik**
Previously the upload was failing in case when local deployment id didn't match the remote deployment id.

### CODE REVIEW

File: aidial_adapter_dial/transformer.py [hunk 1]
```diff
@@ -30,128 +31,121 @@ async def create(
...
+            try:
+                remote_ref_url = await self.get_remote_url(local_ref_url)
+                attachment["reference_url"] = remote_ref_url
+            except Exception:
+                log.error(f"Failed to get remote URL for {local_ref_url!r}")
+
+        if (url := attachment.get("url")) and (
+            local_url := self.local_storage.to_dial_url(url)
+        ):
+            remote_url = await self.get_remote_url(local_url)
```
Comment (roman-romanov-o) [inline aidial_adapter_dial/transformer.py]: Why does `reference_url` wrapped in try/except, but this part doesn't?

File: aidial_adapter_dial/transformer.py [hunk 2]
```diff
@@ -30,128 +31,121 @@ async def create(
...
+        if not local_url.startswith(f"files/{self.local_user_bucket}/"):
             raise ValueError(f"Unexpected local URL: {local_url!r}")
 
-        return f"files/{remote_bucket}/{local_url.removeprefix('files/')}"
+        return f"files/{self.remote_user_bucket}/{local_url.removeprefix('files/')}"
 
-    async def get_local_url(
-        self, remote_url: str, session: aiohttp.ClientSession
-    ) -> str:
+    async def get_local_url(self, remote_url: str) -> str:
```
Comment (roman-romanov-o) [inline aidial_adapter_dial/transformer.py]: You can remove async here

File: aidial_adapter_dial/transformer.py [hunk 3]
```diff
@@ -30,128 +31,121 @@ async def create(
...
             remote_storage=remote_storage,
-            remote_bucket=remote["bucket"],
+            remote_user_bucket=remote["bucket"],
             local_storage=local_storage,
-            local_bucket=local_bucket,
+            local_user_bucket=local_user_bucket,
             local_app_data=local_appdata,
         )
 
     async def get_remote_url(self, local_url: str) -> str:
```
Comment (roman-romanov-o) [inline aidial_adapter_dial/transformer.py]: you can remove async here

File: aidial_adapter_dial/transformer.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_dial/transformer.py]: Indeed. I think we shoudn't shadow the errors and let it fail in **both** cases.

File: aidial_adapter_dial/transformer.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_dial/transformer.py]: Fixed

File: aidial_adapter_dial/transformer.py (hunk 3 above)
Comment (adubovik) [reply aidial_adapter_dial/transformer.py]: Fix

**Review by roman-romanov-o — APPROVED** [review]

**Pull request merged**